plugins/base/src/main/kotlin/transformers/pages/sourcelinks/SourceLinksTransformer.kt

```diff
@@ -134,9 +135,13 @@ class SourceLinksTransformer(val context: DokkaContext) : PageTransformer {
         }
 
     private fun PsiElement.lineNumber(): Int? {
-        val doc = PsiDocumentManager.getInstance(project).getDocument(containingFile)
-        // IJ uses 0-based line-numbers; external source browsers use 1-based
-        return doc?.getLineNumber(textRange.startOffset)?.plus(1)
+        return if (this is SyntheticElement) {
+            null // does not actually exist in code, thus no line number
+        } else {
+            val doc = PsiDocumentManager.getInstance(project).getDocument(containingFile)
+            // IJ uses 0-based line-numbers; external source browsers use 1-based
+            return doc?.getLineNumber(textRange.startOffset)?.plus(1)
```

Comment:
It's not marked as nullable by contract in `PsiElement`, but abstract `LightElement` does return `null` by default. Other than that, usually `TextRange.EMPTY_RANGE` (0, 0) is returned from what I can see.

I thought about checking it for null instead of `SyntheticElement`, but I'm afraid it can lead to bugs where source links are not generated due to some change in the IntelliJ API, and we wouldn't even know, since source links for java projects are not covered by tests at all.

If null is all of a sudden returned here, I personally would prefer it failing loud during testing. The problem was that we don't test on java projects at all, nevermind with configured source links. Alchemist project from the issue is a good candidate for before-release testing, which would've caught it.